packages/http-server-csharp/src/lib/utils.ts

```diff
@@ -755,3 +801,630 @@ export function getCSharpStatusCode(entry: HttpStatusCodesEntry): string | undef
+        return { typeReference: code`${type}`, defaultValue: `${value}`, nullableType: false };
+      case "Tuple":
+        const defaults = [];
+        const [csharpType, isObject] = coalesceTsTypes(program, tsType.values);
+        if (isObject)
+          return { typeReference: "object[]", defaultValue: undefined, nullableType: false };
+        for (const value of tsType.values) {
+          const { defaultValue: itemDefault } = this.getTypeInfo(program, value);
+          defaults.push(itemDefault);
+        }
+        return {
+          typeReference: code`${csharpType.getTypeReference()}[]`,
+          defaultValue: `[${defaults.join(", ")}]`,
+          nullableType: csharpType.isNullable,
+        };
+      case "Object":
+        return { typeReference: code`object`, defaultValue: undefined, nullableType: false };
+      case "Model":
+        let modelResult: EmittedTypeInfo;
+        const cachedResult = this.#anonymousModels.get(tsType);
+        if (cachedResult) {
+          return cachedResult;
+        }
+        if (isRecord(tsType)) {
+          modelResult = { typeReference: code`JsonObject`, nullableType: false };
+        } else {
+          modelResult = {
+            typeReference: code`${this.emitter.emitTypeReference(tsType)}`,
+            nullableType: false,
+          };
+        }
+        this.#anonymousModels.set(tsType, modelResult);
+        return modelResult;
+      case "ModelProperty":
+        return this.getTypeInfo(program, tsType.type);
+      case "Enum":
+        return {
+          typeReference: code`${this.emitter.emitTypeReference(tsType)}`,
+          nullableType: false,
+        };
+      case "EnumMember":
+        if (typeof tsType.value === "number") {
+          const stringValue = tsType.value.toString();
+          if (stringValue.includes(".") || stringValue.includes("e"))
+            return { typeReference: "double", defaultValue: stringValue, nullableType: false };
+          return { typeReference: "int", defaultValue: stringValue, nullableType: false };
+        }
+        if (typeof tsType.value === "string") {
+          return { typeReference: "string", defaultValue: tsType.value, nullableType: false };
+        }
+        return { typeReference: code`object`, nullableType: false };
+      case "Union":
+        return this.getUnionInfo(program, tsType);
+      case "UnionVariant":
+        return this.getTypeInfo(program, tsType.type);
+      default:
+        return {
+          typeReference: code`${this.emitter.emitTypeReference(tsType)}`,
+          nullableType: false,
+        };
+    }
+  }
+  getUnionInfo(program: Program, union: Union): EmittedTypeInfo {
+    const propResult = getNonNullableTsType(union);
+    if (propResult === undefined) {
+      return {
+        typeReference: code`${this.emitter.emitTypeReference(union)}`,
+        nullableType: [...union.variants.values()].filter((v) => isNullType(v.type)).length > 0,
```

Comment:
Nit: Super minor in a sea of changes, but caught my eye as a good use-case for `.some`
```suggestion
        nullableType: [...union.variants.values()].some((v) => isNullType(v.type)),
```